lib/Dialect/FIRRTL/Transforms/IMConstProp.cpp

```diff
@@ -45,6 +46,42 @@ static bool isDeletableWireOrRegOrNode(Operation *op) {
          !hasDontTouch(op) && hasDroppableName(op);
 }
 
+/// This function recursively applies `fn` to leaf ground types of `type`.
+static void
+foreachFIRRTLGroundType(FIRRTLType firrtlType,
+                        llvm::function_ref<void(unsigned, FIRRTLBaseType)> fn) {
+  auto type = firrtlType.dyn_cast<FIRRTLBaseType>();
+  if (!type)
+    type = firrtlType.cast<RefType>().getType();
```

Comment:
FWIW there's a helper for this:
```suggestion
  auto type = getBaseType(firrtlType);
```

Reply:
Yep, thanks!